### PR TITLE
Feature/doughnut fix

### PR DESCRIPTION
**Pull request by AndrewHaslamPF**
Made changes that fixed issues with passing doughnuts:

1. The Doughnut class overloaded the toU8a() method, adding in a length encoding that the plug code was not expecting and resulting in decoding errors. Removing the overloaded method allows encoding to default to the raw implementation, fixing the issue.  This change apparently matches similar changes made by @KarishmaBothara in CENNZNet code.

2. When building the extrinsic signature, an uninitialised doughnut was being created, regardless of whether one was being passed by the caller. Removal of the code that created a default doughnut in ExtrinsicSignature.ts/createPayload() if none was passed fixed the issue.

### CODE REVIEW

**Review by jordy25519 — APPROVED** [review]
`|| createType(this.registry, 'Option<Doughnut>')` I think this was here to add a 0 byte for `Option` i.e. no doughnut. if` doughnut = null` produces the 0 byte then this lgtm!

File: types/ExtrinsicSignature.ts
Comment (jordy25519) [inline types/ExtrinsicSignature.ts]: nit:
```suggestion
      doughnut,
```

Comment (AndrewHaslamPF) [reply types/ExtrinsicSignature.ts]: Applied suggestion in the source branch.

**Review by KarishmaBothara — APPROVED** [review]
lgtm, if it works when no doughnut is attached and extrinsic executes fine..

**Comment by AndrewHaslamPF** [comment]
> lgtm, if it works when no doughnut is attached and extrinsic executes fine..

Yes, works ok with no doughnut attached. 

PS. I can't merge this request - you guys will have to...

**Pull request merged**
